packages/graphiql/src/components/DocExplorer/SearchResults.tsx

```diff
@@ -6,152 +6,123 @@
+  const { schema } = useSchemaContext({ nonNull: true });
+
+  const navItem = explorerNavStack[explorerNavStack.length - 1];
+
+  if (!schema || !navItem.search) {
+    return null;
   }
 
-  render() {
-    const searchValue = this.props.searchValue;
-    const withinType = this.props.withinType;
-    const schema = this.props.schema;
-    const onClickType = this.props.onClickType;
-    const onClickField = this.props.onClickField;
+  const searchValue = navItem.search;
+  const withinType = navItem.def;
+
+  const matchedWithin: ReactNode[] = [];
+  const matchedTypes: ReactNode[] = [];
+  const matchedFields: ReactNode[] = [];
+
+  const typeMap = schema.getTypeMap();
+  let typeNames = Object.keys(typeMap);
 
-    const matchedWithin: ReactNode[] = [];
-    const matchedTypes: ReactNode[] = [];
-    const matchedFields: ReactNode[] = [];
+  // Move the within type name to be the first searched.
+  if (withinType) {
+    typeNames = typeNames.filter(n => n !== withinType.name);
+    typeNames.unshift(withinType.name);
+  }
 
-    const typeMap = schema.getTypeMap();
-    let typeNames = Object.keys(typeMap);
+  for (const typeName of typeNames) {
+    if (
+      matchedWithin.length + matchedTypes.length + matchedFields.length >=
+      100
+    ) {
+      break;
+    }
 
-    // Move the within type name to be the first searched.
-    if (withinType) {
-      typeNames = typeNames.filter(n => n !== withinType.name);
-      typeNames.unshift(withinType.name);
+    const type = typeMap[typeName];
+    if (withinType !== type && isMatch(typeName, searchValue)) {
+      matchedTypes.push(
+        <div className="doc-category-item" key={typeName}>
+          <TypeLink type={type} />
+        </div>,
+      );
     }
 
-    for (const typeName of typeNames) {
-      if (
-        matchedWithin.length + matchedTypes.length + matchedFields.length >=
-        100
-      ) {
-        break;
-      }
-
-      const type = typeMap[typeName];
-      if (withinType !== type && isMatch(typeName, searchValue)) {
-        matchedTypes.push(
-          <div className="doc-category-item" key={typeName}>
-            <TypeLink type={type} onClick={onClickType} />
-          </div>,
-        );
-      }
-
-      if (type && 'getFields' in type) {
-        const fields = type.getFields();
-        Object.keys(fields).forEach(fieldName => {
-          const field = fields[fieldName];
-          let matchingArgs;
-
-          if (!isMatch(fieldName, searchValue)) {
-            if ('args' in field && field.args.length) {
-              matchingArgs = field.args.filter(arg =>
-                isMatch(arg.name, searchValue),
-              );
-              if (matchingArgs.length === 0) {
-                return;
-              }
-            } else {
+    if (type && 'getFields' in type) {
+      const fields = type.getFields();
+      Object.keys(fields).forEach(fieldName => {
+        const field = fields[fieldName];
```

Comment:
Object.entries can make this a bit more elegant.
And according to my tests so far `for of` is always faster than `forEach`

Reply:
Sadly `type.getFields()` does not return an iterator for input object types according to typescript, so we can't use for-of here